### PR TITLE
Add tutorial doc for 3-node in-place upgrade.

### PR DESCRIPTION
**Pull request by jagsgill**
Doc will walk users through the process of configuring/running the playbooks for an in-place upgrade scenario. Based on internal materials from training session.

It is not an exact step-by-step guide since users need to adapt each step to their Gateway environment.

### CODE REVIEW

File: TUTORIAL.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+For example:
+```yaml
+---
+all:
+  hosts:
+  children:
+    gateway:
+      children:
+        gateway_mysql:
```
Comment (emilytzhang) [inline TUTORIAL.md:42]: gateway_mysql moved under gateway now? Is this in-place upgrade host file different from the expedited upgrade one?

File: TUTORIAL.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+                  src: source-failover-db-gw.placeholder.com
+        gateway_processing_node:
+          hosts:
+            source-processing-gw.placeholder.com:
+                src: source-processing-gw.placeholder.com
+```
+
+## Setup Passwords
+
+Enter Gateway passwords in the inventory's `/group_vars/vault` file.
```
Comment (david-dc-chen) [inline TUTORIAL.md:60]: "/group_vars/vault" should be /group_vars/all/vault

File: TUTORIAL.md (hunk 1 above)
Comment (jagsgill) [reply TUTORIAL.md:42]: it is the exact same as `inventories/sample/hosts.yml`, all I say in the doc is to change Gateway host names

File: TUTORIAL.md (hunk 2 above)
Comment (jagsgill) [reply TUTORIAL.md]: fixed

File: TUTORIAL.md (hunk 1 above)
Comment (emilytzhang) [reply TUTORIAL.md:42]: I see I guess this change has merged into develop branch yet.

Comment (jagsgill) [reply TUTORIAL.md:42]: oh yes, this is branched from alister's refactor branch :)

File: TUTORIAL.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+all:
+  hosts:
+  children:
+    gateway:
+      children:
+        gateway_mysql:
+          children:
+            gateway_primary_db:
+              hosts:
+                source-db-gw.placeholder.com:
```
Comment (david-dc-chen) [inline TUTORIAL.md:46]: this example looks outdated, maybe we just let customer reference readme on hosts file?

Comment (jagsgill) [reply TUTORIAL.md:46]: this is intended, for in-place upgrade the target GW = source GW

**Pull request merged**